Failing `Test_scanallips_normalizeStoreInputValue` test

<!-- 
1. Please search to see if an issue already exists for the bug you encountered.
2. For support requests, FAQs or "How to" questions, please use the GitHub Discussions section instead - https://github.com/projectdiscovery/nuclei/discussions or
3. Join our discord server at https://discord.gg/projectdiscovery and post the question on the #nuclei channel.
-->

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### Nuclei version:
dev

### Current Behavior:
```console
--- FAIL: Test_scanallips_normalizeStoreInputValue (20.03s)
    hmap_test.go:101: 
        	Error Trace:	/home/runner/work/nuclei/nuclei/v2/pkg/core/inputs/hybrid/hmap_test.go:101
        	Error:      	elements differ
        	            	
        	            	extra elements in list A:
        	            	([]interface {}) (len=2) {
        	            	 (string) (len=15) "128.199.158.128",
        	            	 (string) (len=23) "[24](https://github.com/projectdiscovery/nuclei/actions/runs/3532199887/jobs/5926263468#step:6:25)00:6180:0:d0::91:1001"
        	            	}
        	            	
        	            	
        	            	extra elements in list B:
        	            	([]interface {}) (len=1) {
        	            	 (string) ""
        	            	}
        	            	
        	            	
        	            	listA:
        	            	([]string) (len=2) {
        	            	 (string) (len=15) "1[28](https://github.com/projectdiscovery/nuclei/actions/runs/3532199887/jobs/5926263468#step:6:29).199.158.128",
        	            	 (string) (len=23) "2400:6180:0:d0::91:1001"
        	            	}
        	            	
        	            	
        	            	listB:
        	            	([]string) (len=1) {
        	            	 (string) ""
        	            	}
        	Test:       	Test_scanallips_normalizeStoreInputValue
        	Messages:   	could not get correct ips
```

### Expected Behavior:
passing test

### Steps To Reproduce:
https://github.com/projectdiscovery/nuclei/actions/runs/3532199868/jobs/5926339275


### Anything else:
https://github.com/projectdiscovery/nuclei/pull/2676#pullrequestreview-1190009029

--- a/v2/pkg/core/inputs/hybrid/hmap_test.go
+++ b/v2/pkg/core/inputs/hybrid/hmap_test.go
@@ -70,7 +70,7 @@ func (this *mockDnsHandler) ServeDNS(w dns.ResponseWriter, r *dns.Msg) {
 			AAAA: net.ParseIP("2400:6180:0:d0::91:1001"),
 		})
 	}
-	w.WriteMsg(&msg)
+	_ = w.WriteMsg(&msg)
 }
 
 func Test_scanallips_normalizeStoreInputValue(t *testing.T) {

--- a/v2/pkg/core/inputs/hybrid/hmap_test.go
+++ b/v2/pkg/core/inputs/hybrid/hmap_test.go
@@ -1,10 +1,13 @@
 package hybrid
 
 import (
+	"net"
 	"os"
+	"strconv"
 	"strings"
 	"testing"
 
+	"github.com/miekg/dns"
 	"github.com/projectdiscovery/hmap/store/hybrid"
 	"github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/contextargs"
 	"github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/protocolstate"
@@ -46,8 +49,41 @@ func Test_expandCIDRInputValue(t *testing.T) {
 	}
 }
 
+type mockDnsHandler struct{}
+
+func (this *mockDnsHandler) ServeDNS(w dns.ResponseWriter, r *dns.Msg) {
+	msg := dns.Msg{}
+	msg.SetReply(r)
+	switch r.Question[0].Qtype {
+	case dns.TypeA:
+		msg.Authoritative = true
+		domain := msg.Question[0].Name
+		msg.Answer = append(msg.Answer, &dns.A{
+			Hdr: dns.RR_Header{Name: domain, Rrtype: dns.TypeA, Class: dns.ClassINET, Ttl: 60},
+			A:   net.ParseIP("128.199.158.128"),
+		})
+	case dns.TypeAAAA:
+		msg.Authoritative = true
+		domain := msg.Question[0].Name
+		msg.Answer = append(msg.Answer, &dns.AAAA{
+			Hdr:  dns.RR_Header{Name: domain, Rrtype: dns.TypeA, Class: dns.ClassINET, Ttl: 60},
+			AAAA: net.ParseIP("2400:6180:0:d0::91:1001"),
+		})
+	}
+	w.WriteMsg(&msg)
+}
+
 func Test_scanallips_normalizeStoreInputValue(t *testing.T) {
+	srv := &dns.Server{Addr: ":" + strconv.Itoa(61234), Net: "udp"}
+	srv.Handler = &mockDnsHandler{}
+
+	go func() {
+		err := srv.ListenAndServe()
+		require.Nil(t, err)
+	}()
+
 	defaultOpts := types.DefaultOptions()
+	defaultOpts.InternalResolversList = []string{"127.0.0.1:61234"}
 	_ = protocolstate.Init(defaultOpts)
 	tests := []struct {
 		hostname string

